modules/openapi-generator/src/main/resources/r/api.mustache

```diff
@@ -635,11 +633,58 @@
         if (is.null(local_var_resp$response) || local_var_resp$response == "") {
           local_var_resp$response <- "API server error"
         }
-        local_var_resp
+        return(local_var_resp)
         {{/returnExceptionOnFailure}}
       }
     }{{^-last}},{{/-last}}
     {{/operation}}
+  ),
+  private = list(
+    #' @description
+    #' Write response to a file
+    #' 
+    #' The function will write out data. 
+    #' 
+    #' 1. If binary data is detected it will use `writeBin`
+    #' 2. If the raw response is coercable to text, the text will be written to a file
+    #' 3. If the raw response is not coercable to text, the raw response will be written
+    #' 
+    #' @param local_var_resp The API response
+    #' @param file The name of the data file to save the result
+    WriteFile = function(local_var_resp, file) {
+      if (private$IsBinary(local_var_resp$response)) {
+        writeBin(local_var_resp$response, file)
+      } else {
+        response <- private$Deserialize(local_var_resp)
+        base::write(response, file)
+      }
+    },
+
+    #' @description
+    #' Check response for binary content
+    #' 
+    #' @param local_var_resp The API response
+    IsBinary = function(x) {
+      # ref: https://stackoverflow.com/a/17098690/1785752
+      b <- readBin(x, "int", n = 1000, size=1, signed=FALSE)
+      return(max(b) > 128)
+    },
+
+    #' @description
+    #' Deserialize the response
+    #' 
+    #' @param local_var_resp The API response
+    #' @param return_type The target return type for the endpoint (e.g., `"object"`). If `NULL` text will be left as-is.
+    #' @return If the raw response is corecable to text, return the text. Otherwise return the raw resposne.
+    Deserialize = function(local_var_resp, return_type = NULL) {
+      text <- local_var_resp$ResponseAsText()
+      if (is.na(text)) {
+        return(local_var_resp$response)
+      } else if (is.null(return_type)) {
+        return(text)
+      }
+      return(self$api_client$deserialize(text, return_type, loadNamespace("{{packageName}}")))
+    }
```

Comment:
what about adding these functions to api_client (which contains other utility functions)?

otherwise we will be repeating the same set of functions (private) in every auto-generated API files

Reply:
Done. Note that the code will be appear once in generated clients, but will need to be maintained in 2 places:

1. `modules/openapi-generator/src/main/resources/r/api_client.mustache`
2. `modules/openapi-generator/src/main/resources/r/libraries/httr2/api_client.mustache`